rust/datafusion/tests/func_provider.rs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::sync::Arc;
+
+use arrow::datatypes::DataType;
+use arrow::util::display::array_value_to_string;
+
+use datafusion::error::Result;
+use datafusion::execution::context::ExecutionContext;
+use datafusion::physical_plan::functions::{ActualFunction, FuncProvider};
+use datafusion::physical_plan::functions::{ReturnTypeFunction, Signature};
+use datafusion::physical_plan::udf::ScalarUDF;
+
+use datafusion::physical_plan::ColumnarValue;
+use datafusion::scalar::ScalarValue;
+
+#[derive(Debug, Clone)]
+pub struct DatabaseExec {
+    db_name: String,
+}
+
+impl DatabaseExec {
+    pub fn try_new(db_name: &str) -> Result<Self> {
+        Ok(Self {
+            db_name: db_name.to_string(),
+        })
+    }
+}
+
+impl FuncProvider for DatabaseExec {
+    fn evaluate(&self, inputs: &[ColumnarValue]) -> Result<ColumnarValue> {
+        Ok(ColumnarValue::Scalar(ScalarValue::Utf8(Some(
+            self.db_name.to_string(),
+        ))))
+    }
+}
+
+async fn assert_provider_row_count(expected_count: String) -> Result<()> {
```

Comment:
the `expected_count` name here is confusing to me -- as it is a string. Perhaps `expected_name`?

Reply:
yes, expected_name is right